src-docs/src/views/html_id_generator/use_generated_html_id.js

```diff
@@ -0,0 +1,41 @@
+import React, { useState, Fragment } from 'react';
+
+import {
+  EuiSwitch,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiSpacer,
+  EuiCode,
+  EuiFormRow,
+} from '../../../../src/components';
+
+import { useGeneratedHtmlId } from '../../../../src/services';
+
+export const UseGeneratedHtmlId = () => {
+  const generatedId = useGeneratedHtmlId({ prefix: 'Some', suffix: 'id' });
+
+  const [isChecked, setIsChecked] = useState(false);
+  const onChange = (e) => setIsChecked(e.target.checked);
+
+  return (
+    <Fragment>
+      <EuiFlexGroup
+        justifyContent="flexStart"
+        gutterSize="m"
+        alignItems="center"
+      >
+        <EuiFlexItem grow={false}>
+          <EuiFormRow>
+            <EuiSwitch
+              label="Clicking me changes component state"
+              checked={isChecked}
+              onChange={onChange}
+            />
+          </EuiFormRow>
+        </EuiFlexItem>
+      </EuiFlexGroup>
```

Comment:
NIT, but you don't need flex groups for just a single flex item unless you're trying to vertically center. And the form row does nothing either without any props.

```suggestion
       <EuiSwitch
         label="Clicking me changes component state"
         checked={isChecked}
         onChange={onChange}
       />
```

That will clean up the demo code to get to the heart of the example.

Reply:
https://github.com/elastic/eui/pull/5133/commits/0a2a20a11522f7edcb72533c3d031588d60d89e9